### ISSUE TITLE
In `library`, update signature of `control item` components to be consistant

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#1314]
# Context

With the 4 control-item-based components, signatures are not aligned:
```swift
OUDSCheckboxItem(isOn: $isOn, label: "Checkbox item")
OUDSCheckboxItemIndeterminate(selection: $selection, label: "Checkbox item")
OUDSRadioItem(isOn: $isOn, label: "Radio button item")
OUDSSwitchItem("Switch item", isOn: $isOn) // Best solution
```

We can notice here for the `OUDSSwitchItem` we have first the label, not named, and then the binding *isOn*, to be aligned with what brings the native SwiftUI API.
However for the `OUDSCheckboxItem` and the `OUDSRadioItem` the *isOn* binding is used first, and then the named label.
Things should be aligned.

# Definition of Done

- [x] Deprecate old signatures(s) and add new signature(s) (first label then binding) for *checkbox item*
- [x] Deprecate old signatures(s) and add new signature(s) (first label then binding) for *checkbox item indeterminate*
- [x] Deprecate old signatures(s) and add new signature(s) (first label then binding) for *switch item*
- [x] Deprecate old signatures(s) and add new signature(s) (first label then binding) for *radio item*
- [x] Update AGENTS.md for the 4 components
- [x] Update tutorial for the 4 components
- [x] Update Swift DocC documentation for the 4 components
- [x] Update the design system toolbox app (code samples) for the 4 components

# Notes

- As the public API won't be broken, no need to update the MIGRATION file nor increment the major version of the lib
- The label parameter must be unnamed and in first position
- The binding must be in second position after the label

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.3.0**
- Build number: **1771625071**
- Build tag: **bf3c57d**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by B3nz01d** [comment]
Issue finished during Sprint # 28

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.3.0**
- Build number: **1773698781**
- Build tag: **acc2a2e**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.3.0**
- Build number: **1773785054**
- Build tag: **fdcd323**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.3.0**
- Build number: **1773871478**
- Build tag: **37ddea2**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.3.0**
- Build number: **1774389878**
- Build tag: **770a249**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New release available** 🚀

Please, get the [new release from GitHub](https://github.com/Orange-OpenSource/ouds-ios-design-system-toolbox/releases/tag/1.3.0) and wait for App Store update to test it!

- Display name: **Design System Toolbox**
- Version: **1.3.0**
- Build number: **1774540892**



_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_
